automation/src/test/scala/org/broadinstitute/dsde/firecloud/test/library/DataLibrarySpec.scala

```diff
@@ -51,4 +51,58 @@ class DataLibrarySpec extends FreeSpec with WebBrowserSpec with UserFixtures wit
       }
     }
   }
+
+  /**
+    * This test creates two workspace associated with different tags. Search for tag should
+    *  only find right workspace in Data Library table.
+    */
+  "Use a known tag to search for dataset" in withWebDriver { implicit driver =>
+    val curatorUser = UserPool.chooseCurator
+    implicit val authToken: AuthToken = curatorUser.makeAuthToken()
+    withWorkspace(namespace, "DataLibrarySpec_tags_", attributes = Some(WorkspaceData.tagA)) { wsName1 =>
+      withWorkspace(namespace, "DataLibrarySpec_tags_", attributes = Some(WorkspaceData.tagB)) { wsName2 =>
+        withCleanUp {
+
+          val tag1: String = WorkspaceData.tagA.get("tag:tags").get(0)
+          val tag2: String = WorkspaceData.tagB.get("tag:tags").get(0)
+          val data1 = LibraryData.metadataBasic + ("library:datasetName" -> wsName1)
+          api.library.setLibraryAttributes(namespace, wsName1, data1)
+          val data2 = LibraryData.metadataBasic + ("library:datasetName" -> wsName2)
+          api.library.setLibraryAttributes(namespace, wsName2, data2)
+          register cleanUp {
+            api.library.unpublishWorkspace(namespace, wsName1)(authToken)
+            api.library.unpublishWorkspace(namespace, wsName2)(authToken)
+          }
+          api.library.publishWorkspace(namespace, wsName1)
+          api.library.publishWorkspace(namespace, wsName2)
+
+          withSignIn(curatorUser) { _ =>
+            val page = new DataLibraryPage().open
+
+            // search by tag1 alone
+            val aRows: List[Map[String, String]] = page.doTagsSearch(tag1)
+            // check: wsName1 and tag1 should be found in table
+            aRows.exists { row => row("Cohort Name") == wsName1 && row("Tags").contains(tag1) } shouldBe true
```

Comment:
Ideally, these would use `should contain` instead of `shouldBe true`. That's a little tricky in this case because of the queries on a list of maps but it's worth a try. Check out http://www.scalatest.org/user_guide/using_matchers.